src/test/java/org/opensearch/knn/training/VectorReaderTests.java

```diff
@@ -66,59 +64,100 @@ public void testRead_valid_completeIndex() throws InterruptedException, Executio
 
         // Read all vectors and confirm they match vectors
         TestVectorConsumer testVectorConsumer = new TestVectorConsumer();
-        final CountDownLatch inProgressLatch1 = new CountDownLatch(1);
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
         vectorReader.read(
             clusterService,
-            indexName,
-            fieldName,
-            10000,
-            10,
+            DEFAULT_INDEX_NAME,
+            DEFAULT_FIELD_NAME,
+            DEFAULT_MAX_VECTOR_COUNT,
+            DEFAULT_SEARCH_SIZE,
             testVectorConsumer,
-            ActionListener.wrap(response -> inProgressLatch1.countDown(), e -> fail(e.toString()))
+            createOnSearchResponseCountDownListener(inProgressLatch)
         );
 
-        assertTrue(inProgressLatch1.await(100, TimeUnit.SECONDS));
+        assertLatchDecremented(inProgressLatch);
 
         List<Float[]> consumedVectors = testVectorConsumer.getVectorsConsumed();
-        assertEquals(numVectors, consumedVectors.size());
+        assertEquals(DEFAULT_NUM_VECTORS, consumedVectors.size());
 
         List<Float> flatVectors = vectors.stream().flatMap(Arrays::stream).collect(Collectors.toList());
         List<Float> flatConsumedVectors = consumedVectors.stream().flatMap(Arrays::stream).collect(Collectors.toList());
         assertEquals(new HashSet<>(flatVectors), new HashSet<>(flatConsumedVectors));
     }
 
+    public void testRead_valid_trainVectorsIngestedAsIntegers() throws IOException, ExecutionException, InterruptedException {
+        createIndex(DEFAULT_INDEX_NAME);
+        createKnnIndexMapping(DEFAULT_INDEX_NAME, DEFAULT_FIELD_NAME, DEFAULT_DIMENSION);
+
+        // Create list of random vectors and ingest
+        Random random = new Random();
+        List<Integer[]> vectors = new ArrayList<>();
+        for (int i = 0; i < DEFAULT_NUM_VECTORS; i++) {
+            Integer[] vector = new Integer[DEFAULT_DIMENSION];
+
+            for (int j = 0; j < DEFAULT_DIMENSION; j++) {
+                vector[j] = random.nextInt();
+            }
```

Comment:
nit: you can do the same but with less code via streams:

`
Integer[] array = random.ints(size, lowBound, highBound).boxed().toArray(Integer[]::new);
`

Reply:
Thanks will update